NitroxLauncher/Patching/AppHelper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Security.Principal;
+using System.Windows;
+
+namespace NitroxLauncher
+{
+    public class AppHelper
+    {
+        public static string ProgramFileDirectory = Environment.ExpandEnvironmentVariables("%ProgramW6432%");
+
+        public static bool IsAppRunningInAdmin()
+        {
+            WindowsPrincipal wp = new WindowsPrincipal(WindowsIdentity.GetCurrent());
+            return wp.IsInRole(WindowsBuiltInRole.Administrator);
+        }
+
+        public static void RestartAsAdmin(string path)
```

Comment:
Could you separate the logic from this method out into two?
`RestartAsAdmin` sounds good if it only restarted the app but it does more than that: checks the path of program files. 

You can remove the comment in the `SubnauticaPath` property if you change the code to something like:
```cs
if (!AppHelper.IsRunningAsAdmin() && Path.GetFullPath(path).StartsWith(ProgramFileDirectory, StringComparison.OrdinalIgnoreCase))
{
    AppHelper.RestartAsAdmin();
}
```